pdata/pcommon/map.go

```diff
@@ -55,8 +55,7 @@ func (m Map) EnsureCapacity(capacity int) {
 // It is allowed to modify the returned value using Value.Set* functions.
 // Such modification will be applied to the value stored in this map.
 //
-// If the key does not exist returns an invalid instance of the KeyValue and false.
-// Calling any functions on the returned invalid instance will cause a panic.
+// If the key does not exist returns a zero-initialized KeyValue and false.
```

Comment:
I have removed "Calling any functions on the returned invalid instance will cause a panic." because not all functions cause a panic. Methods on the value have relevant comments, so we can leave for the user to check docs per method instead of listing them here

Reply:
To make the risks immediately clear, how about 

`Calling any functions on the returned invalid instance may cause a panic.`

Reply:
Thanks I have added that comment!